[Security Solution] Event Renderer Virtualization

logeekal opened the pull request:
## Summary

This PR implements virtualization when Event Renderers are enabled. Ideally from UX pespective nothing should change but from performance perspective, the event renderers should be scalable.

### Testing checklist

1. UX is working same as before when Event Renderers are enabled.
2. Operations such as increasing page size from `10` to `100` are not taking as much time as before. Below operations can be used to test.
   a. Closing / Opening Timeline
   b. Changes `Rows per page`
   c. Changes tabs from query to any other and back.

### Before
In below video, you will notice how long it took to change `pageSize` to 100 and all 100 rows are rendered at once.

https://github.com/user-attachments/assets/106669c9-bda8-4b7d-af3f-b64824bde397


### After

https://github.com/user-attachments/assets/356d9e1f-caf1-4f88-9223-0e563939bf6b



> [!Note]
> 1. Also test in small screen. The table should be scrollable but nothing out of ordinary.
> 2. Additionally, try to load data which has `network_flow` process so as to create bigger and varied Event Renderers.


<!--ONMERGE {"backportTargets":["8.x"]} ONMERGE-->

logeekal committed:
fix: virt. working with height issue

logeekal committed:
fix: working list virt - react-virtualized

logeekal committed:
react-window virtualization

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!

logeekal committed:
incremental changes

janmonschke commented:
https://github.com/user-attachments/assets/895226c8-ccdb-420b-9bcf-f40522ff362f

Overall this seems to be working :)

I noticed two small issues:

1. The vertical scrollbar for the event rendered items is hidden. I've tested this with multiple screen widths and it's always outside of the container. (see attached video)
2. Closing a timeline with 500+ items takes a loooooong time. This might not be an issue with this PR but with timeline in general 🤷

logeekal commented:
> I noticed two small issues:
> 
>   >  The vertical scrollbar for the event rendered items is hidden. I've tested this with multiple screen widths and it's always outside of the container. (see attached video)
>   >  Closing a timeline with 500+ items takes a loooooong time. This might not be an issue with this PR but with timeline in general 🤷
> 

Hey @janmonschke 

1. First is a bug in EUI for which i have raised a bug here : https://github.com/elastic/eui/issues/8038
2. I am checking if closing timeline performance is related to this PR.

logeekal committed:
Great pert with auto height.

Avoid height calculation for row renderer, leads to great performance.
This was done by passing `auto` height.

logeekal committed:
fix: working great + better performance

logeekal committed:
fix: fast and working RC

logeekal commented:
@janmonschke ,

>>The vertical scrollbar for the event rendered items is hidden. I've tested this with multiple screen widths and it's always outside of the container. (see attached video)

>>Closing a timeline with 500+ items takes a loooooong time. This might not be an issue with this PR but with timeline in general 🤷

Both of these things are fixed and table should feel much faster than before. Could you please take another look?

logeekal committed:
correct deps

janmonschke commented:
@logeekal Just tested it and both issues are gone, wohooo 🎉 

I noticed another thing where the table stays blank for quite a while when scrolling. This feels like it comes from the virtualization and I wonder if it performs better when React is not running in dev mode. Do you see the same?

https://github.com/user-attachments/assets/bb69f90f-baa1-4829-9c27-dfa54921f268

logeekal committed:
eui_rc

janmonschke commented:
Thanks @logeekal. I tested it again and this time around it was much faster. Not sure what was happening on my machine earlier. The simple rows render much faster than the event renderers but both feel fast enough.

kqualters-elastic commented on x-pack/plugins/security_solution/public/timelines/components/timeline/unified_components/data_table/custom_timeline_data_grid_body.tsx:
```diff
@@ -42,31 +54,137 @@ const DEFAULT_UDT_ROW_HEIGHT = 34;
+    }, []);
+
+    const innerRowContainer = useMemo(() => {
+      const InnerComp = React.forwardRef<HTMLDivElement, PropsWithChildren<{}>>(
```
can probably pull this definition to the top level of this file, and that way this is never recreated.

logeekal replied:
yes.. I was just waiting for the EUI teams opinion and then i will refactor code to be put in production. Currently, this is just working code. 

I wanted to get everyone opinions mainly on performance. But what you say makes sense.

kqualters-elastic replied:
Understood, I like the changes, ultimately though I think to finally get the performance to where we really want, we will need to continue to refactor on the eui side. Specifically, useRowHeightUtils and all the associated hooks can likely be cleaned up, especially for use in data_grid_body_custom.tsx, where virtualization is not used at all.

logeekal replied:
@kqualters-elastic , I just checked, it has depenendencies `headerRow` and `footerRow` and hence it cannot be pulled out.

cee-chen committed:
Upgrade EUI to v97.0.0

cee-chen committed:
i18n updates

cee-chen committed:
[EuiDataGrid] Handle breaking `renderCustomGridBody` change for Alerts table

cee-chen committed:
[EuiDataGrid] Handle breaking `renderCustomGridBody` change for Security timeline table

cee-chen committed:
[EuiLink] Update snapshots to account for new SR text copy

cee-chen committed:
[EuiLink] Update text assertions/selectors to account for new SR text copy

cee-chen committed:
[EuiLink] Update test regex utils to account for new SR text copy

cee-chen committed:
[EuiButtonDisplay] Update snapshots

cee-chen committed:
[EuiLink] Update test icon assertion to account for the fact the icon no longer has text

- text is now entirely SR only

logeekal committed:
Merge remote-tracking branch 'cee/eui/v97.0.0' into feat/custom_grid_body_list_virtualization

logeekal committed:
fix: more optims

logeekal committed:
remove unnecessary logic

logeekal commented:
/ci

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/241629)
* Commit: 7e65a8745ba62356830209383dd54d0487c0f786

### Failed CI Steps
* [Checks](https://buildkite.com/elastic/kibana-pull-request/builds/241629#01927bd0-9a22-49d4-8aad-2b31d391c35b)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241629","number":241629,"commit":"7e65a8745ba62356830209383dd54d0487c0f786"}],"number":241629}
buildkite-pr-comment-->

logeekal committed:
fix: jest tests

logeekal committed:
Merge branch 'main' into feat/custom_grid_body_list_virtualization

logeekal committed:
chore: remove unnecessary console statements

elasticmachine commented:
Pinging @elastic/security-threat-hunting-investigations (Team:Threat Hunting:Investigations)

janmonschke commented:
I am seeing some vertical alignment glitches when scrolling. But only for the initial scroll. I wonder if this goes away in react prod mode. Wdyt @logeekal?


https://github.com/user-attachments/assets/ef77a670-f7db-4085-989e-24c5c2e26a94

logeekal commented:
>I am seeing some vertical alignment glitches when scrolling. But only for the initial scroll. I wonder if this goes away in react prod mode. Wdyt @logeekal?

## Issue and Cause
1. This is not related to this PR since Event Renderer virtualization is implemented in Timeline Code where Event Renderers occurs as a complete Row.

2. This is related to this PR ( https://github.com/elastic/kibana/pull/192827) where I enabled Alert Table Virtualization. You can also observe this issue in Discover when running in Dev Mode as show below. 

   - I think this issue occurs for 2 reasons: Performance of calculation of height of EUI data grid and rendering performance of our own Renderer.

https://github.com/user-attachments/assets/0569318b-dbb7-4b9e-b750-dec918ccdc01

If you check same Alert Table in `Prod` mode in SIEM dev, it is performing `mostly` fine as show below:


https://github.com/user-attachments/assets/b0623f51-987a-40ef-bd97-7496fc772deb

### Solution

IMO as a long term solution for this, we need to improve the rendering performance of our `Row Renderers`. For which I am anyways going to raise an issue and work on.

PS: I will repeat that this issue is not related to this PR. This PR scope is only limited to timeline.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/242429)
* Commit: 7399da6ebf04e562cc5f366ff993efe3060a5990

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/242429#01928f12-d2ef-4820-b098-cae6ca7776b0)

### History
* :broken_heart: [Build #241629](https://buildkite.com/elastic/kibana-pull-request/builds/241629) failed 7e65a8745ba62356830209383dd54d0487c0f786

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242429","number":242429,"commit":"7399da6ebf04e562cc5f366ff993efe3060a5990"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241629","number":241629,"commit":"7e65a8745ba62356830209383dd54d0487c0f786"}],"number":242429}
buildkite-pr-comment-->

janmonschke commented:
Gotcha, so this issue of the alert table not expanding to full height is also not part of this PR then?

<img width="2009" alt="Screenshot 2024-10-15 at 11 05 06" src="https://github.com/user-attachments/assets/15b7455e-3ed4-4d10-8a49-1f84b0ab4559">

logeekal commented:
>Gotcha, so this issue of the alert table not expanding to full height is also not part of this PR then?

No. The alert table height was intentionally made limited to enable virtualization because otherwise performance was pretty bad. This was done in PR : https://github.com/elastic/kibana/pull/192827

logeekal committed:
ci: 😊

logeekal commented:
/ci

janmonschke commented on x-pack/plugins/security_solution/public/timelines/components/timeline/tabs/shared/use_timeline_control_columns.tsx:
```diff
@@ -84,49 +138,7 @@ export const useTimelineControlColumn = ({
-            />
-          );
-        },
+        rowCellRender: JEST_ENVIRONMENT ? RowCellRender : React.memo(RowCellRender),
```
I'm not a huge fan of checking this here. Could you elaborate in a comment here on why it's necessary to not memoize in test mode?

logeekal replied:
if we do not do it, we get below warning in JEST tests. EUI is not happy when a memoized component is passed.

![grafik](https://github.com/user-attachments/assets/eae7062e-b4d7-4c0f-bc15-ad3f66a47a4b)

janmonschke replied:
if you could say that in a comment right there, that would be perfect 👍 
Plus, we should probably open up an issue with EUI about this. Sounds like one of their prop types is wrong.

janmonschke commented on x-pack/plugins/security_solution/public/timelines/components/timeline/unified_components/data_table/custom_timeline_data_grid_body.tsx:
```diff
@@ -217,6 +377,9 @@ const CustomDataGridSingleRow = memo(function CustomDataGridSingleRow(
       {canShowRowRenderer ? (
         <Cell
+          rowHeightsOptions={defaultAutoHeight}
+          /* @ts-expect-error because currently CellProps do not allow string width but it is important to be passed for height calculations   */
+          width={'100%'}
```
So it currently "works" because the rest props are spread onto the component?

logeekal replied:
Exactly. Here is the code in EUI: https://github.com/elastic/eui/pull/8028/files#diff-79459b26a5c68fbbf1233bd815d9f57ba81326b9632b27b3ddb3da6b0a25c92eR229

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/242598)
* Commit: 5c089b033aca74c5e16de7fe0df7e94ec21cba3c
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-193316-5c089b033aca`

### Failed CI Steps


### History
* :broken_heart: [Build #242429](https://buildkite.com/elastic/kibana-pull-request/builds/242429) failed 7399da6ebf04e562cc5f366ff993efe3060a5990
* :broken_heart: [Build #241629](https://buildkite.com/elastic/kibana-pull-request/builds/241629) failed 7e65a8745ba62356830209383dd54d0487c0f786

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242598","number":242598,"commit":"5c089b033aca74c5e16de7fe0df7e94ec21cba3c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242429","number":242429,"commit":"7399da6ebf04e562cc5f366ff993efe3060a5990"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241629","number":241629,"commit":"7e65a8745ba62356830209383dd54d0487c0f786"}],"number":242598}
buildkite-pr-comment-->

logeekal committed:
Merge branch 'main' into feat/custom_grid_body_list_virtualization

logeekal committed:
fix: cypress test

logeekal committed:
fix: remove focused test

nkhristinin commented on x-pack/test/security_solution_cypress/cypress/e2e/detection_response/detection_engine/rule_creation/indicator_match_rule.cy.ts:
```diff
@@ -525,14 +525,9 @@ describe('indicator match', { tags: ['@ess', '@serverless', '@skipInServerlessMK
-          }${
+          `${getNewThreatIndicatorRule().threat_mapping[0].entries[0].field}matched${
             indicatorRuleMatchingDoc.atomic
-          }${accessibilityText}threat.enrichments.matched.typeindicator_match_rule${accessibilityText}provided` +
```
Hey, why we remove this accessibilityText?

logeekal replied:
The accessibility text was related to draggables in Event Renderers and they are no longer draggables. I have created a follow-up issue to add a new text for all Event Renderers and not just this : https://github.com/elastic/kibana/issues/196505

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/242980)
* Commit: a64ccc4bba20ed594108f86acfb672a92ae7c40d
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-193316-a64ccc4bba20`

### History
* :broken_heart: [Build #242598](https://buildkite.com/elastic/kibana-pull-request/builds/242598) failed 5c089b033aca74c5e16de7fe0df7e94ec21cba3c
* :broken_heart: [Build #242429](https://buildkite.com/elastic/kibana-pull-request/builds/242429) failed 7399da6ebf04e562cc5f366ff993efe3060a5990
* :broken_heart: [Build #241629](https://buildkite.com/elastic/kibana-pull-request/builds/241629) failed 7e65a8745ba62356830209383dd54d0487c0f786

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242980","number":242980,"commit":"a64ccc4bba20ed594108f86acfb672a92ae7c40d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242598","number":242598,"commit":"5c089b033aca74c5e16de7fe0df7e94ec21cba3c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242429","number":242429,"commit":"7399da6ebf04e562cc5f366ff993efe3060a5990"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241629","number":241629,"commit":"7e65a8745ba62356830209383dd54d0487c0f786"}],"number":242980}
buildkite-pr-comment-->

logeekal committed:
fix: types

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/243012)
* Commit: fd9e1ef537faad9c2cb7d0749ae1fd3801c37d4f
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-193316-fd9e1ef537fa`

### History
* :broken_heart: [Build #242980](https://buildkite.com/elastic/kibana-pull-request/builds/242980) failed a64ccc4bba20ed594108f86acfb672a92ae7c40d
* :broken_heart: [Build #242598](https://buildkite.com/elastic/kibana-pull-request/builds/242598) failed 5c089b033aca74c5e16de7fe0df7e94ec21cba3c
* :broken_heart: [Build #242429](https://buildkite.com/elastic/kibana-pull-request/builds/242429) failed 7399da6ebf04e562cc5f366ff993efe3060a5990
* :broken_heart: [Build #241629](https://buildkite.com/elastic/kibana-pull-request/builds/241629) failed 7e65a8745ba62356830209383dd54d0487c0f786

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/243012","number":243012,"commit":"fd9e1ef537faad9c2cb7d0749ae1fd3801c37d4f"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242980","number":242980,"commit":"a64ccc4bba20ed594108f86acfb672a92ae7c40d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242598","number":242598,"commit":"5c089b033aca74c5e16de7fe0df7e94ec21cba3c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242429","number":242429,"commit":"7399da6ebf04e562cc5f366ff993efe3060a5990"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241629","number":241629,"commit":"7e65a8745ba62356830209383dd54d0487c0f786"}],"number":243012}
buildkite-pr-comment-->

logeekal committed:
Merge branch 'main' into feat/custom_grid_body_list_virtualization

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/243048)
* Commit: fcdbaf0a49ad7eb64bfcf66b6b0b9d7779fc659f
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-193316-fcdbaf0a49ad`

### History
* :broken_heart: [Build #243012](https://buildkite.com/elastic/kibana-pull-request/builds/243012) failed fd9e1ef537faad9c2cb7d0749ae1fd3801c37d4f
* :broken_heart: [Build #242980](https://buildkite.com/elastic/kibana-pull-request/builds/242980) failed a64ccc4bba20ed594108f86acfb672a92ae7c40d
* :broken_heart: [Build #242598](https://buildkite.com/elastic/kibana-pull-request/builds/242598) failed 5c089b033aca74c5e16de7fe0df7e94ec21cba3c
* :broken_heart: [Build #242429](https://buildkite.com/elastic/kibana-pull-request/builds/242429) failed 7399da6ebf04e562cc5f366ff993efe3060a5990
* :broken_heart: [Build #241629](https://buildkite.com/elastic/kibana-pull-request/builds/241629) failed 7e65a8745ba62356830209383dd54d0487c0f786

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/243048","number":243048,"commit":"fcdbaf0a49ad7eb64bfcf66b6b0b9d7779fc659f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/243012","number":243012,"commit":"fd9e1ef537faad9c2cb7d0749ae1fd3801c37d4f"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242980","number":242980,"commit":"a64ccc4bba20ed594108f86acfb672a92ae7c40d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242598","number":242598,"commit":"5c089b033aca74c5e16de7fe0df7e94ec21cba3c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242429","number":242429,"commit":"7399da6ebf04e562cc5f366ff993efe3060a5990"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241629","number":241629,"commit":"7e65a8745ba62356830209383dd54d0487c0f786"}],"number":243048}
buildkite-pr-comment-->

logeekal committed:
Merge branch 'main' into feat/custom_grid_body_list_virtualization

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/243106)
* Commit: 86e5d00853a59741a103cfa22a20698fbf60e3a7
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-193316-86e5d00853a5`

### History
* :broken_heart: [Build #243048](https://buildkite.com/elastic/kibana-pull-request/builds/243048) failed fcdbaf0a49ad7eb64bfcf66b6b0b9d7779fc659f
* :broken_heart: [Build #243012](https://buildkite.com/elastic/kibana-pull-request/builds/243012) failed fd9e1ef537faad9c2cb7d0749ae1fd3801c37d4f
* :broken_heart: [Build #242980](https://buildkite.com/elastic/kibana-pull-request/builds/242980) failed a64ccc4bba20ed594108f86acfb672a92ae7c40d
* :broken_heart: [Build #242598](https://buildkite.com/elastic/kibana-pull-request/builds/242598) failed 5c089b033aca74c5e16de7fe0df7e94ec21cba3c
* :broken_heart: [Build #242429](https://buildkite.com/elastic/kibana-pull-request/builds/242429) failed 7399da6ebf04e562cc5f366ff993efe3060a5990

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/243106","number":243106,"commit":"86e5d00853a59741a103cfa22a20698fbf60e3a7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/243048","number":243048,"commit":"fcdbaf0a49ad7eb64bfcf66b6b0b9d7779fc659f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/243012","number":243012,"commit":"fd9e1ef537faad9c2cb7d0749ae1fd3801c37d4f"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242980","number":242980,"commit":"a64ccc4bba20ed594108f86acfb672a92ae7c40d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242598","number":242598,"commit":"5c089b033aca74c5e16de7fe0df7e94ec21cba3c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242429","number":242429,"commit":"7399da6ebf04e562cc5f366ff993efe3060a5990"}],"number":243106}
buildkite-pr-comment-->

logeekal committed:
Merge branch 'main' into feat/custom_grid_body_list_virtualization

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/243193)
* Commit: ecd76537d461abda4f7371ac75032d235a7a16c9
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-193316-ecd76537d461`

### Failed CI Steps
* [x-pack/test_serverless/functional/test_suites/observability/common_configs/config.group6.ts](https://buildkite.com/elastic/kibana-pull-request/builds/243193#01929630-205e-4807-921c-5379d186d199)
* [FTR Configs #8](https://buildkite.com/elastic/kibana-pull-request/builds/243193#01929630-207d-420b-8373-bf8342ed8b74)

### History
* :broken_heart: [Build #243106](https://buildkite.com/elastic/kibana-pull-request/builds/243106) failed 86e5d00853a59741a103cfa22a20698fbf60e3a7
* :broken_heart: [Build #243048](https://buildkite.com/elastic/kibana-pull-request/builds/243048) failed fcdbaf0a49ad7eb64bfcf66b6b0b9d7779fc659f
* :broken_heart: [Build #243012](https://buildkite.com/elastic/kibana-pull-request/builds/243012) failed fd9e1ef537faad9c2cb7d0749ae1fd3801c37d4f
* :broken_heart: [Build #242980](https://buildkite.com/elastic/kibana-pull-request/builds/242980) failed a64ccc4bba20ed594108f86acfb672a92ae7c40d
* :broken_heart: [Build #242598](https://buildkite.com/elastic/kibana-pull-request/builds/242598) failed 5c089b033aca74c5e16de7fe0df7e94ec21cba3c

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/243193","number":243193,"commit":"ecd76537d461abda4f7371ac75032d235a7a16c9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/243106","number":243106,"commit":"86e5d00853a59741a103cfa22a20698fbf60e3a7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/243048","number":243048,"commit":"fcdbaf0a49ad7eb64bfcf66b6b0b9d7779fc659f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/243012","number":243012,"commit":"fd9e1ef537faad9c2cb7d0749ae1fd3801c37d4f"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242980","number":242980,"commit":"a64ccc4bba20ed594108f86acfb672a92ae7c40d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242598","number":242598,"commit":"5c089b033aca74c5e16de7fe0df7e94ec21cba3c"}],"number":243193}
buildkite-pr-comment-->

kibanamachine commented:
Starting backport for target branches: 8.x

https://github.com/elastic/kibana/actions/runs/11371137966

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/196587">](https://github.com/elastic/kibana/pull/196587)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 9.4.3 -->
